### PR TITLE
Revert "[release-16.0] Replace use of `WAIT_UNTIL_SQL_THREAD_AFTER_GTIDS` with `WAIT_FOR_EXECUTED_GTID_SET` (#14612)"

### PR DESCRIPTION
**Pull request by deepthi**
Reverts vitessio/vitess#14620

because it causes #14738

### CODE REVIEW

**Review by rohit-nayak-ps — APPROVED** [review]

**Comment by deepthi** [comment]
Cluster(12) passed on `push` but failed on `pull_request`. Looks like flakiness. We can override and merge after approval.

**Comment by rohit-nayak-ps** [comment]
> `Cluster(12)` passed on `push` but failed on `pull_request`. Looks like flakiness. We can override and merge after approval.

That "success" was because the workflow was skipped lol! Checked because it said finished in 3 seconds.

Added a Do Not Merge for now, it has failed thrice in succession and I am retrying once more. Issue is unlikely related to this PR, but might be worth a look if it fails again. 

@GuptaManan100 FYI.

```
2023-12-09T20:41:24.4276312Z I1209 20:36:27.114068   19013 vtctlclient_process.go:214] Executing vtctlclient with command: vtctlclient --server localhost:16206 PlannedReparentShard -- --keyspace_shard test_keyspace/0 --wait_replicas_timeout 31s --new_primary zone1-361 (attempt 1 of 10)
2023-12-09T20:41:24.4280092Z E1209 20:36:58.216089   19013 vtctlclient_process.go:71] error in PlannedReparentShard output E1209 20:36:58.212578   21854 main.go:96] E1209 20:36:58.212061 planned_reparenter.go:710] some replicas failed to reparent; retry PlannedReparentShard with the same new primary alias ([zone1-0000000361 tablet zone1-0000000362 failed to SetReplicationSource(zone1-0000000361): rpc error: code = DeadlineExceeded desc = context deadline exceeded: rpc error: code = DeadlineExceeded desc = context deadline exceeded]) to retry failed replicas: %!v(MISSING)
2023-12-09T20:41:24.4284306Z PlannedReparentShard Error: rpc error: code = Unknown desc = some replicas failed to reparent; retry PlannedReparentShard with the same new primary alias (zone1-0000000361) to retry failed replicas: tablet zone1-0000000362 failed to SetReplicationSource(zone1-0000000361): rpc error: code = DeadlineExceeded desc = context deadline exceeded: rpc error: code = DeadlineExceeded desc = context deadline exceeded
2023-12-09T20:41:24.4288017Z E1209 20:36:58.214874   21854 main.go:105] remote error: rpc error: code = Unknown desc = some replicas failed to reparent; retry PlannedReparentShard with the same new primary alias (zone1-0000000361) to retry failed replicas: tablet zone1-0000000362 failed to SetReplicationSource(zone1-0000000361): rpc error: code = DeadlineExceeded desc = context deadline exceeded: rpc error: code = DeadlineExceeded desc = context deadline exceeded
2023-12
```

**Review by mattlord — APPROVED** [review]

**Comment by GuptaManan100** [comment]
Looking at the test failure.

**Pull request merged**
